src/librbd/Operations.cc

```diff
@@ -800,7 +801,7 @@ int Operations<I>::snap_rollback(const cls::rbd::SnapshotNamespace& snap_namespa
 
     r = prepare_image_update(false);
     if (r < 0) {
-      return -EROFS;
+      return r;
```

Comment:
@dillaman When cherry-picking e8eee15, in `Operations<I>::update_features`, you did not update the case like this  because it was not backported at that moment. But it has been backported just recently and I think you need to update your cherry-pick.

[1] https://github.com/ceph/ceph/pull/24122

Reply:
Rebased and fixed -- thx